python/ray/tune/trainable/trainable.py

```diff
@@ -1030,15 +1026,22 @@ def restore_from_object(self, obj):
             "Use Trainable.restore() instead."
         )
 
-    def delete_checkpoint(self, checkpoint_path: Union[str, Checkpoint]):
+    def delete_checkpoint(
+        self, checkpoint_path: Union[str, LegacyCheckpoint, NewCheckpoint]
+    ):
```

Comment:
Let's hard deprecate this API. Was only needed to schedule remote deletions, but that happens via the storage filesystem directly now.